pallet/nft/src/impls.rs

```
@@ -376,6 +377,58 @@ impl<T: Config> Pallet<T> {
 		(new_cursor, total_owned, response)
 	}
 
+	/// Find the tokens details for the given collection id
+	/// Returns collection owner, name, metadata schema, max issuance,
+	/// next available serial number, collection issuance, is_cross_chain_compatible
+	pub fn collection_details(collection_id: CollectionUuid) -> CollectionDetail<T::AccountId>
+	where
+		<T as frame_system::Config>::AccountId: core::default::Default,
+	{
+		let collection_info = <CollectionInfo<T>>::get(collection_id);
+		if collection_info.is_none() {
+			// should not happen
```

There is a chance that an invalid collection_id was supplied to the RPC so this could be None, we should handle this situation and return an error/ None instead of Default